sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -253,3 +253,19 @@ TEST(Context, MatchingKeys)
   value = valueT3.Get<int>();
   EXPECT_TRUE(value == 456);
 }
+
+struct SomeStructForContext : ValueBase
```

Comment:
Does it have to derive from `ValueBase`?

Reply:
Yes, I got surprised as well. But this is what adds a little type safety to the type.
ValueBase defines a contract of an object that can be destructed.
Then, when we get add the object to the Context, we can only add the ValueBase, and lose all the info about the derived class.
Anyone getting it from the Context needs to parse it to retrieve the derived class.

It is almost as ugly as `void*`,  almost, but still not a raw void*.